x/epochstorage/keeper/stake_storage.go

```diff
@@ -119,17 +130,19 @@ func (k Keeper) removeAllEntriesPriorToBlockNumber(ctx sdk.Context, block uint64
 	for _, chainId := range allChainID {
 		for _, entry := range allStorage {
 			if strings.Contains(entry.Index, chainId) {
-				if (len(chainId)) > len(entry.Index) {
-					panic(fmt.Sprintf("storageType + chainId length out of range %d vs %d\n more info: entry.Index: %s, chainId: %s", len(chainId), len(entry.Index), entry.Index, chainId))
-				}
 				storageBlock := entry.Index[:(len(entry.Index) - len(chainId))]
 				blockHeight, err := strconv.ParseUint(storageBlock, 10, 64)
 				if err != nil {
 					if storageBlock == "" {
 						// if storageBlock is empty its stake entry current. so we dont remove it.
 						continue
 					}
-					panic("failed to convert storage block to int: " + storageBlock)
+					utils.LavaFormatError("critical: failed to decode storage block", err,
```

Comment:
we said this panic can stay since this function is only used for testing

Reply:
right ... restore the panic and moved the function to testing code.
fixed 4b66dc3c